clerk.go

```diff
@@ -448,6 +448,32 @@ type ClientConfig struct {
 	BackendConfig
 }
 
+// Clock is an interface that can be used with the library instead
+// of the [time] package.
```

Comment:
To the reviewer: The `Clock` interface is not a drop-in replacement for the [time](https://pkg.go.dev/time) package.

I've only defined one method on `Clock`, `Now()`, because this is the only function of the `time` package that our library currently needs.

I'm not certain whether we should keep the interface small, or require that it implements all of the `time` package's available functions, even if we don't currently need them.

The alternative would be to have this interface for `clerk.Clock`:
```go
type Clock interface {
	After(d time.Duration) <-chan time.Time
	Sleep(d time.Duration)
	Now() time.Time
	Since(t time.Time) time.Duration
	NewTicker(d time.Duration) Ticker
	NewTimer(d time.Duration) Timer
	AfterFunc(d time.Duration, f func()) Timer
}
```

Let me know what you think.

Reply:
I feel like the `Now()` method is enough for what we are trying to achieve here, expose a way to DI the clock dependency for testing purposes and not only